src/wovnio/html/HtmlConverter.php

```diff
@@ -276,10 +276,31 @@ private function insertHreflangTags($html)
         return $this->insertAfterTag($parent_tags, $html, implode('', $hreflangTags));
     }
 
+    private function translateCanonicalTag($html)
+    {
+        if ($this->isNoindexLang($this->headers->requestLang())) {
+            return $html;
+        }
+
+        $canonical_tag_regex = "/<link[^>]*rel=\"canonical\"*[^>]*href=\"(.*)\"\>/";
```

Comment:
`href=\"(.*)\"` should be `href=\"[^\"]*)\" I think. I'm seeing it match weirdly.

Reply:
Yeah you are right, it does look wrong.